stories/src/virtual/react-virtuoso/board.tsx

```diff
@@ -0,0 +1,287 @@
+import type { Quote, QuoteMap } from '../../types';
+import QuoteCountSlider from '../quote-count-chooser';
+
+const Container = styled.div`
+  display: flex;
+`;
+
+interface RowProps {
+  quote: Quote;
+  provided: DraggableProvided;
+  isDragging: boolean;
+}
+
+// Memoizing row items for even better performance!
+const Row = React.memo(({ quote, provided, isDragging }: RowProps) => {
+  return (
+    <div
+      {...provided.draggableProps}
+      {...provided.dragHandleProps}
+      ref={provided.innerRef}
+      style={provided.draggableProps.style}
+      className={`item ${isDragging ? 'is-dragging' : ''}`}
+    >
+      <div style={{ padding: '4px 8px 4px 8px' }}>
+        <QuoteItem
+          quote={quote}
+          isDragging={isDragging}
+          style={{ margin: '0px' }}
+        />
+      </div>
+    </div>
+  );
+});
+
+const HeightPreservingItem = ({ children, ...props }: any) => {
+  const [size, setSize] = useState(0);
+  const knownSize = props['data-known-size'];
+  useEffect(() => {
+    setSize((prevSize) => {
+      return knownSize === 0 ? prevSize : knownSize;
+    });
+  }, [knownSize]);
+
+  return (
+    <div
+      {...props}
+      className="height-preserving-container"
+      // check styling in the style tag below
+      style={{ '--child-height': `${size}px` }}
+    >
+      {children}
+    </div>
+  );
+};
+
+interface ColumnProps {
+  columnId: string;
+  quotes: Quote[];
+}
+
+const ColumnContainer = styled.div`
+  border-top-left-radius: ${borderRadius}px;
+  border-top-right-radius: ${borderRadius}px;
+  background-color: ${colors.N30};
+  flex-shrink: 0;
+  margin: ${grid}px;
+  display: flex;
+  flex-direction: column;
+`;
+
+const Column = React.memo(function Column(props: ColumnProps) {
+  const { columnId, quotes } = props;
+
+  return (
+    <ColumnContainer>
+      <Title>{columnId}</Title>
+      <Droppable
+        droppableId={columnId}
+        mode="virtual"
+        renderClone={(
+          provided: DraggableProvided,
+          snapshot: DraggableStateSnapshot,
+          rubric: DraggableRubric,
+        ) => (
+          <Row
+            provided={provided}
+            isDragging={snapshot.isDragging}
+            quote={quotes[rubric.source.index]}
+          />
+        )}
+      >
+        {(droppableProvided, snapshot) => {
+          // Add an extra item to our list to make space for a dragging item
+          // Usually the DroppableProvided.placeholder does this, but that won't
+          // work in a virtual list
+          const itemCount: number = snapshot.isUsingPlaceholder
+            ? quotes.length + 1
+            : quotes.length;
+
+          return (
+            <Virtuoso
+              components={{
+                Item: HeightPreservingItem,
+              }}
+              // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+              // @ts-ignore
+              scrollerRef={droppableProvided.innerRef}
```

Comment:
We should avoid the use of `@ts-ignore` when we can. Instead of by passing `TypeScript` we could make sure that we are getting what we are expecting.

In this case, `droppableProvided.innerRef` is expecting an `HTMLElement` or `null`, but `scrollerRef` can also give the `window` object. This causes an error, because based on `droppableProvided.innerRef`'s type it can't take the `window` object as argument. We can make this type error go away if we add a check like we do in this following suggeston:

```suggestion
              scrollerRef={(scroller) => {
                invariant(
                  scroller instanceof HTMLElement || scroller === null,
                  'scroller should be an instance of HTMLElement or be null',
                );

                droppableProvided.innerRef(scroller);
              }}
```